app/src/main/java/com/limelight/Game.java

```diff
@@ -2264,7 +2264,16 @@ public void onUsbPermissionPromptCompleted() {
     public boolean onKey(View view, int keyCode, KeyEvent keyEvent) {
         switch (keyEvent.getAction()) {
             case KeyEvent.ACTION_DOWN:
-                return handleKeyDown(keyEvent);
+                boolean handled = handleKeyDown(keyEvent);
```

Comment:
`handleKeyDown()` should already be handling `KEYCODE_BACK` and trying to determine if it's from a gamepad (where it is sent to the PC as the Select/Back button) or a remote or other input device (where it finishes the activity). In the former case (which is the common case), this back menu won't be accessible.

What's the case where you're seeing errant stream exits?

Reply:
In handleKeyDown the navigation keys are just forwarded to the system.

I moved the opening of the GameMenu inside handleKeyDown to make it better understandable :-).